src/ui/SearchInterface/SearchInterface.ts

```diff
@@ -1230,10 +1214,38 @@ export class StandaloneSearchInterface extends SearchInterface {
     // By using a setTimeout, we allow other possible code related to the search box / magic box time to complete.
```

Comment:
On line 1212, there is a potential undesirable behaviour in that if there are no encoded hash values, we will have a trailing `&` or `#`.

I did not fix it to keep the PR scoped, and because it's very unlikely to have no hash values. At a minimum, I expect there will be at least one UA event which will appear as the `firstQueryCause` hash value.